src/helpers/useAuth.tsx

```diff
@@ -24,10 +29,47 @@ let AuthContext = createContext<Context>({
 
 export function Provider(props: Props) {
   let [token, setToken] = useState<string>('');
+  let { data: userData } = useGetAuthenticatedUser();
+  let expiresAt = new Date(userData?.authenticatedUser.expiresAt || '');
+  let now = new Date();
+
+  now.setDate(now.getDate() + 1);
+
+  let { setUser } = useSetAuthenticatedUser();
+
+  let { renewToken } = useCustomerRenewToken({
+    variables: { customerAccessToken: token },
+    onCompleted: ({ customerAccessTokenRenew }) => {
+      if (
+        customerAccessTokenRenew &&
+        customerAccessTokenRenew.customerAccessToken
+      ) {
+        let {
+          accessToken,
+          expiresAt,
+        } = customerAccessTokenRenew.customerAccessToken;
+        setToken(accessToken);
+        if (userData && userData) {
+          setUser({
+            variables: {
+              user: {
+                ...userData?.authenticatedUser,
+                expiresAt: expiresAt,
+              },
+            },
+          });
+        }
+      }
+    },
+    onError: () => {},
+  });
 
   useEffect(() => {
     getToken().then((token) => {
       // TODO: Check token expiration date
+      if (expiresAt < now) {
+        renewToken();
+      }
       setToken(token || '');
     });
   }, []);
```

Comment:
and add this, the error should be gone
```suggestion
  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```